pandas/core/arrays/datetimes.py

```diff
@@ -1298,6 +1298,31 @@ def isocalendar(self):
             iso_calendar_df.iloc[self._isnan] = None
         return iso_calendar_df
 
+    @property
+    def weekofyear(self):
+        """
+        The week ordinal of the year.
+
+        .. deprecated:: 1.1.0
+
+        weekofyear and week have been deprecated.
+        Please use DatetimeIndex.isocalendar().week instead.
+        """
+        import pandas as pd
+
+        warnings.warn(
+            "weekofyear and week have been deprecated, please use "
+            "DatetimeIndex.isocalendar().week instead, which returns "
+            "a Series.  To exactly reproduce the behavior of week and "
+            "weekofyear and return an Index, you may call "
+            "pd.Int64Index(idx.isocalendar().week)",
+            FutureWarning,
+            stacklevel=3,
+        )
+        return pd.Int64Index(self.isocalendar().week)
```

Comment:
this should return ndarray here and be wrapped on DatetimeIndex to return Int64Index

Reply:
Of course, thanks.